pkg/k8s/factory_functions.go

```diff
@@ -154,3 +175,331 @@ func copyObjToV1Namespace(obj interface{}) meta_v1.Object {
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV1Services(o1, o2 interface{}) bool {
+	_, ok := o1.(*v1.Service)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Service", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*v1.Service)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Service", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV1Endpoints(o1, o2 interface{}) bool {
+	_, ok := o1.(*v1.Endpoints)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Endpoints", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*v1.Endpoints)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Endpoints", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV1beta1Ingress(o1, o2 interface{}) bool {
+	_, ok := o1.(*v1beta1.Ingress)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1beta1.Ingress", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*v1beta1.Ingress)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1beta1.Ingress", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV2CNP(o1, o2 interface{}) bool {
+	_, ok := o1.(*cilium_v2.CiliumNetworkPolicy)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *cilium_v2.CiliumNetworkPolicy", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*cilium_v2.CiliumNetworkPolicy)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *cilium_v2.CiliumNetworkPolicy", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV1Pod(o1, o2 interface{}) bool {
+	_, ok := o1.(*v1.Pod)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Pod", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*v1.Pod)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Pod", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV1Node(o1, o2 interface{}) bool {
+	_, ok := o1.(*v1.Node)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Node", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*v1.Node)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Node", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
+	return false
+}
+
+func equalV1Namespace(o1, o2 interface{}) bool {
+	_, ok := o1.(*v1.Namespace)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Namespace", reflect.TypeOf(o1))
+		return false
+	}
+	_, ok = o1.(*v1.Namespace)
+	if !ok {
+		log.Panicf("Invalid resource type %q, expecting *v1.Namespace", reflect.TypeOf(o2))
+		return false
+	}
+	// FIXME write dedicated deep equal function
```

Comment:
What are the implications of missing these proper equals implementations? Is this code that is currently unused but will be used and fixed up by your other outstanding PRs?

Reply:
https://github.com/cilium/cilium/pull/5449#issue-212469985

>In following PRs the dedicated Equalness and MissingFunc should be written for each object. Without those functions the default behavior of kubernetes watcher remains the same.
>
>An example of the of those dedicated function implementations can be found in the last commit of #5519